### PR TITLE
validate time partition keys when adding to a TimeWindowPartitionsSubset

### PR DESCRIPTION
**Pull request by OwenKephart**
### Summary & Motivation

Align with this comment: https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/storage/partition_status_cache.py?L157&subtree=true

### CODE REVIEW

**Review by sryza — COMMENTED** [review]
Two thoughts:
- My instinct would be that `with_partition_keys` should error in this case.
- Do we know the perf impact of this?

**Comment by OwenKephart** [comment]
perf impact should be fairly minimal, as this gets called on all partition keys in other parts of the code, and has never been an issue. agree that this feels a bit weird to allow -- I'd also be down to move. this logic to the spot indicated by the linked comment if you think that'd be better (probably adding an is_valid_partition_key to TimeWindowPartitionsDefinition)

**Comment by sryza** [comment]
I think the spot with the comment is better, at the very least for consistency

File: python_modules/dagster/dagster/_core/definitions/time_window_partitions.py [hunk 1]
```diff
@@ -1337,6 +1344,7 @@ def _add_partitions_to_time_windows(
         return result_windows, num_added_partitions
 
     def with_partition_keys(self, partition_keys: Iterable[str]) -> "TimeWindowPartitionsSubset":
+
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/time_window_partitions.py]: Remove this

**Review by sryza — APPROVED** [review]
This looks good to me. I think it's worth @clairelin135 taking a look too.

**Pull request merged**